addons/account/models/account_move_line.py

```diff
@@ -1057,12 +1057,15 @@ def _compute_is_refund(self):
             if line.move_id.move_type in ('out_refund', 'in_refund'):
                 is_refund = True
             elif line.move_id.move_type == 'entry':
-                taxes = line.tax_repartition_line_id.tax_id or line.tax_ids[:1]
-                tax_type = set(taxes.mapped('type_tax_use'))
-                if tax_type == {'sale'} and line.credit == 0:
-                    is_refund = True
-                elif tax_type == {'purchase'} and line.debit == 0:
-                    is_refund = True
+                if line.tax_repartition_line_id:
+                    is_refund = line.tax_repartition_line_id.document_type == 'refund'
+                else:
+                    taxes = line.tax_repartition_line_id.tax_id or line.tax_ids[:1]
+                    tax_type = set(taxes.mapped('type_tax_use'))
+                    if tax_type == {'sale'} and line.credit == 0:
+                        is_refund = True
+                    elif tax_type == {'purchase'} and line.debit == 0:
+                        is_refund = True
```

Comment:
Can you play on the tax_tag_invert value instead? You should be able to set the value when creating the move lines for the statement line in the bank reco widget

Reply:
I tried that, but it always ends up being recomputed.